### ISSUE TITLE
eslint/no-restricted-imports false negative

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#10984]
given the following, the module graph appears to be empty

```
-------- source --------

import "foo"; import "./index.mjs"; import "index";
-------- rule config --------
[
  {
    "patterns": [
      {
        "group": [
          "[@a-z]*",
          "!.*/**",
          "./index.mjs"
        ],
        "message": "foo is forbidden, use bar instead"
      }
    ]
  }
]
```

same for 

```
import {} from "foo"
```
